File: app/src/main/java/com/permissionnanny/dagger/AppComponent.java
package com.permissionnanny.dagger;

import android.app.Application;
import com.permissionnanny.data.AppPermissionManager;
import com.permissionnanny.data.OngoingRequestDB;
import dagger.Component;

import javax.inject.Singleton;

/**
 *
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    Application app();

    AppModule.Bus bus();

    AppPermissionManager appPermissionManager();

    OngoingRequestDB db();
}

File: app/src/main/kotlin/com/permissionnanny/dagger/AppComponent.kt
package com.permissionnanny.dagger

import android.app.Application
import com.permissionnanny.data.AppPermissionManager
import com.permissionnanny.data.OngoingRequestDB
import dagger.Component

import javax.inject.Singleton

/**

 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun app(): Application

    fun bus(): AppModule.Bus

    fun appPermissionManager(): AppPermissionManager

    fun db(): OngoingRequestDB
}
